extending/provider/implementation.rst

```diff
@@ -1,85 +1,44 @@
 Implementation
 --------------
 
-Annotations
-+++++++++++
+Every cloud provider is made up of one or more resources. Each resource implementation will manage the lifecycle
+of a resource in the cloud: create, read, update, and delete.
 
-**@Type(string)**
-    The class should be annotated with the ``@Type(string)`` annotation. The name provided by this annotation is
-    used by the Gyro language to lookup the resource implementation. For example, ``@Type("instance")`` in the AWS
-    provider will make ``aws::instance`` available to the Gyro language.
+A resource is any entity in a cloud provider that has state. For example, ``aws::load-balancer`` or
+``aws::instance`` are resources with state where-as a service like AWS Rekognition has no persistent state so
+it would not need to be implemented as a resource in Gyro.
 
-    The Java package(s) that make up a provider should be annotated with one or more of the following:
+With the exception of ``refresh()`` all CRUD methods provide a ``GyroUI`` and ``State`` objects.
 
-**@Namespace(string)**
-    This is name the primary namespace of the provider that will be exposed to the Gyro language. For example, for
-    the resource ``aws::instance`` this annotation defines the ``aws`` portion of the resource name. The namespace
-    should be short, all-lower case, unique, and concisely describe the provider.
+The ``GyroUI`` objects can be used to output additional information to the user's console.
 
-    This is a ``package`` annotation intended to be used in a ``package-info.java`` file at the root of the provider
-    projects main package.
+The ``State`` object can be used to periodically save state between operations within one of the CRUD methods. For
+example, if creation of an object requires multiple API calls then state should be saved (i.e. ``state.save()``) between
+each API call to ensure state has a view consistent with the cloud provider's state.
 
-**@DocNamespace(string)**
-    Same as above but used for auto-generated documentation.
+Implementing refresh
+====================
 
-**@DocGroup(string)**
-    As explained above each provider should be logically separated into packages based on the provider's API and
-    service groupings. Each of these service group packages should define this annotation.
+The ``refresh()`` method is called to refresh the state of a resource. Gyro will call this method on every
```

Comment:
Should you say up front the dual purposes ``refresh()`` serves. It's first purpose to determine if the resource already exists in the cloud and returning ``true`` if so. Secondly, assuming the resource is found in the cloud, as its namesake implies to refresh a Java instance of the resource with cloud values so Gyro can compare against its own state.